packages/verified-fetch/src/plugins/plugin-handle-tar.ts

```diff
@@ -0,0 +1,39 @@
+import { code as dagPbCode } from '@ipld/dag-pb'
+import toBrowserReadableStream from 'it-to-browser-readablestream'
+import { code as rawCode } from 'multiformats/codecs/raw'
+import { tarStream } from '../utils/get-tar-stream.js'
+import { notAcceptableResponse, okResponse } from '../utils/responses.js'
+import { BasePlugin } from './plugin-base.js'
+import type { PluginContext } from './types.js'
+
+/**
+ * Accepts a UnixFS `CID` and returns a `.tar` file containing the file or
+ * directory structure referenced by the `CID`.
+ */
+export class TarPlugin extends BasePlugin {
+  readonly codes = []
+  canHandle ({ cid, accept }: PluginContext): boolean {
+    this.log('checking if we can handle %c with accept %s', cid, accept)
+    return accept === 'application/x-tar'
+  }
+
+  async handle (context: PluginContext): Promise<Response> {
+    const { cid, path, resource, options } = context
```

Comment:
Is pathing supported? should we be using `pathDetails.terminalElement`? https://github.com/ipfs/specs/issues/488